src/util/http_utils.js

```diff
@@ -298,6 +299,10 @@ async function parse_request_body(req, options) {
         try {
             const res = querystring.parse(req.body.toString());
             const renamed = _.mapKeys(res, (value, key) => _.snakeCase(key));
+            // If x-amz-content-sha256 isn't provided, we calculate it ourselves -
+            if (!req.headers['x-amz-content-sha256']) {
+                req.body_sha256_calc_by_nb = AWS.util.crypto.sha256(req.body, 'hex');
```

Comment:
Hmm, so I see that we already calculate the incoming content sha256 in the end of `read_request_body()` and we save it in `req.content_sha256_buf` so this calculation is actually redundant in any case. 

I am not sure if this fits all cases but it seems that you could just do this generally in read_request_body:
```
                req.content_sha256_buf = sha256_buf;
                req.content_sha256_sig ||= sha256_buf;
```